roles/wildfly_systemd/tasks/service.yml

```diff
@@ -11,4 +11,12 @@
     state: "{{ instance_state }}"
     enabled: "{{ wildfly_systemd.enabled | default(omit) }}"
     daemon_reload: true
+  register: wildfly_service_status
   become: yes
+
+- name: Check if PID directory exists
+  ansible.builtin.stat:
+    path: "{{ wildfly_systemd.selinux.pid_path }}"
+  register: pid_directory
+  notify: "Check that PID ownership are correct"
+  when: wildfly_service_status.status.ActiveState == "active"
```

Comment:
That's not really what I meant. I think the handlers should be called when the role wildfly_systemd changes the state of the service. So the "call to the handler" would not lived in this role.

@guidograzioli WDYT?

Reply:
but if the user, except from`wildfly_install` and `wildfly_systemd` roles, doesn't add any other, then probably this will never work.

Reply:
yeah, so the directory will always exist because, either it already exist or we create it. The pidfile, we can check if it exists after starting the service. Maybe more a jws_validation task than jws_systemd bu your choice

Reply:
and of course s/jws/wildfly

Reply:
@guidograzioli @rpelisse I have modified the PR. Now it'll check if the PID directory path exists or not, second when wildfly service runs then it'll automatically create PID file inside the directory. After that, It'll validate if the PID file is created or not. Meanwhile, if someone changes `wildfly_pidfile_homedir` value or deletes the PID file, then it'll restart wildfly service, which will regenerate the PID file.